packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenModule.kt

```diff
@@ -15,10 +15,20 @@ class SplashScreenOptions : Record {
 }
 
 class SplashScreenModule : Module() {
+  private var _userControlledAutoHideEnabled: Boolean = false
+
   override fun definition() = ModuleDefinition {
     Name("ExpoSplashScreen")
 
     AsyncFunction<Unit>("preventAutoHideAsync") {
+      // The user has manually invoked prevent autohide, this is used to allow libraries
+      // such as expo-router to know whether it's safe to hide or if they should wait for
+      // the user to do it.
+      _userControlledAutoHideEnabled = true
+      SplashScreenManager.preventAutoHideCalled = true
+    }
+
+    AsyncFunction<Unit>("_internal_preventAutoHideAsync") {
```

Comment:
```suggestion
    AsyncFunction<Unit>("internalPreventAutoHideAsync") {
```
Because users can't reference the module directly,  we can use normal naming conventions
```typescript
export async function _internal_preventAutoHideAsync(): Promise<boolean> {
  if (!SplashModule) {
    return false;
  }

  return SplashModule.internalPreventAutoHideAsync();
}
```